src/extension.ts

```diff
@@ -1737,8 +1747,34 @@ async function deleteContextKubernetes(explorerNode: explorer.KubernetesObject)
 }
 
 async function useNamespaceKubernetes(explorerNode: explorer.KubernetesObject) {
-    if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
-        refreshExplorer();
+    if (explorerNode) {
+        if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
+            refreshExplorer();
+        }
+    } else {
+        const currentNS = await kubectlUtils.currentNamespace(kubectl);
+        promptKindName([kuberesources.allKinds.namespace], undefined,
+            {
+                prompt: 'What namespace do you want to use?',
+                placeHolder: 'Enter the namespace to switch to or press enter to select from available list',
+                filterNames: [currentNS]
+            },
+            async (resource) => {
+                if (resource) {
```

Comment:
use the early return pattern as described above to return early if no resource was provided.

Reply:
i guess thats fall under the else part and should return automatically there. We can skip early return here.

Reply:
@kameshsampath I'm not sure I understand your reasoning.

Reply:
i meant to say for the early return in this case might not be needed as there no more lines post that which needs to be evaluated, since if the `resource` is not there then it skip the block and return automatically.

Reply:
ah got it. early returns are more style than function. i prefer to use them since they don't use as much (unpleasant, in my opinion) indentation, providing a more clear picture to the reader as to what's going on in the truthy condition in functions which only perform one thing or the other. 

the following are exactly the same functionally:

```
function myFunction() {
    if (someTruthyValue) {
         something();
    } else {
         somethingElse();
    }
}
```

vs.
 
```
function myFunction {
    if (!someTruthyValue) {
        somethingElse();
        return;
    }

    something();
}
```

the truthy branch has no impact on the non-truthy branch.